controller/ProviderAdmin.php

```diff
@@ -41,4 +44,32 @@ public function getClassService()
     {
         return $this->getServiceLocator()->get(RdfLtiProviderRepository::class);
     }
+
+    protected function getExtraValidationRules(): array
+    {
+        return $this->getValidationFactory()->createFormValidators($this->getLtiVersion());
+    }
+
+    private function getValidationFactory(): ValidatorsFactory
+    {
+        /** @noinspection PhpIncompatibleReturnTypeInspection */
+        return $this->getServiceLocator()->get(ValidatorsFactory::SERVICE_ID);
+    }
+
+    private function getConfigurationMapper(): LtiProviderFieldsMapper
+    {
+        /** @noinspection PhpIncompatibleReturnTypeInspection */
```

Comment:
Why?

Reply:
That is optional. To make PHPStorm not so annoying about types compatibility, as I noticed a lot of people started doing the same

TLDR; 
 `ServiceLocatorInterface::get` phpdoc annotation says `object|array`, but that is not so compatible with strongly specified `LtiProviderFieldsMapper`.